src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -983,16 +991,30 @@ public void updateControlledGcsBucket(
    * @param updateParams resource properties to update
    */
   public void updateReferencedBigQueryDataTable(
-      UUID workspaceId, UUID resourceId, UpdateResourceParams updateParams) {
+      UUID workspaceId, UUID resourceId, UpdateReferencedBqTableParams updateParams) {
     // convert the CLI object to a WSM request object
-    UpdateDataReferenceRequestBody updateRequest =
-        new UpdateDataReferenceRequestBody()
-            .name(updateParams.name)
-            .description(updateParams.description);
+    UpdateBigQueryDataTableReferenceRequestBody updateRequest =
+        new UpdateBigQueryDataTableReferenceRequestBody()
+            .name(updateParams.getResourceParams().name)
+            .description(updateParams.getResourceParams().description);
+    if (updateParams.updateReferenceTarget()) {
+      updateRequest.resourceAttributes(
+          new GcpBigQueryDataTableAttributes()
+              .projectId(
+                  Optional.ofNullable(updateParams.getProjectId())
+                      .orElse(updateParams.getOriginalResource().getProjectId()))
+              .datasetId(
+                  Optional.ofNullable(updateParams.getDatasetId())
+                      .orElse(updateParams.getOriginalResource().getDatasetId()))
```

Comment:
I think WSM API should be changed to accept `null` to mean no-change for the update endpoints. How does WSM team feel about making this change, or is there already a ticket to do that?

Reply:
filed a ticket: https://broadworkbench.atlassian.net/browse/PF-1271